azure-keyvault/azure/keyvault/keys/_client.py

```diff
@@ -1,114 +1,144 @@
+from typing import Any, Dict, Generator, Mapping, Optional, List
+from datetime import datetime
 import uuid
 
+from azure.core.configuration import Configuration
+from azure.core.pipeline.policies import (
+    UserAgentPolicy,
+    RetryPolicy,
+    RedirectPolicy,)
+from azure.core.pipeline.transport import RequestsTransport, HttpRequest, HttpResponse
+from azure.core.pipeline import Pipeline
+from azure.core.exceptions import ClientRequestError
+from azure.keyvault._internal import _BearerTokenCredentialPolicy
+
+from .._generated import DESERIALIZE, SERIALIZE
 from ._models import (
+    Key,
     DeletedKey,
+    KeyBase,
+)
+from .._generated.v7_0.models import (
     DeletedKeyItem,
     DeletedKeyItemPaged,
-    JsonWebKey,
-    Key,
-    KeyAttributes,
     KeyCreateParameters,
-    KeyItem,
     KeyItemPaged,
+    KeyRestoreParameters,
     KeyUpdateParameters,
 )
-from azure.core.configuration import Configuration
-from azure.core.exceptions import ClientRequestError
-from azure.core.pipeline import Pipeline
-from azure.core.pipeline.policies import (
-    HTTPPolicy,
-    UserAgentPolicy,
-    HeadersPolicy,
-    RetryPolicy,
-    RedirectPolicy,
-    ContentDecodePolicy,
-)
-from azure.core.pipeline.transport import (
-    RequestsTransport,
-    HttpRequest,
-    HttpResponse,
-    HttpTransport,
-)
-from msrest import Serializer, Deserializer
-
-
-class BearerTokenCredentialPolicy(HTTPPolicy):
-    def __init__(self, credentials):
-        self._credentials = credentials
-
-    def send(self, request, **kwargs):
-        # type: (HttpRequest, Any) -> HttpResponse
-        auth_header = "Bearer " + self._credentials.token["access_token"]
-        request.http_request.headers["Authorization"] = auth_header
-
-        return self.next.send(request, **kwargs)
+from .._generated.v7_0.models import KeyAttributes as _KeyAttributes
 
 
 class KeyClient:
-    API_VERSION = "7.0"
+    _api_version = "7.0"
 
     @staticmethod
     def create_config(**kwargs):
         # type: (Any) -> Configuration
         config = Configuration(**kwargs)
-        config.user_agent = UserAgentPolicy("KeyClient", **kwargs)
+        config.user_agent = UserAgentPolicy('KeyClient', **kwargs)
         config.headers = None
         config.retry = RetryPolicy(**kwargs)
         config.redirect = RedirectPolicy(**kwargs)
 
-        # TODO: these are requests-specific
-        config.cert = config.timeout = None
-        config.verify = True
         return config
 
-    def __init__(self, vault_url, credentials, config=None, transport=None):
-        # type: (str, Any, Configuration, HttpTransport) -> None
-        self.vault_url = vault_url.strip("/")
-        config = config or KeyClient.create_config()
-        transport = RequestsTransport(config)
+    def __init__(self, vault_url, credentials, config=None, transport=None, **kwargs):
+        # type: (str, Any, Configuration, HttpTransport, Mapping[str, Any]) -> None
```

Comment:
The type here won't change with azure-identity.  This package won't reference azure-identity.  The type of credentials should be TokenCredential from azure-core

---
In reply to: [284858517](https://github.com/Azure/azure-sdk-for-python/pull/5184#discussion_r284858517) [](ancestors = 284858517)

Reply:
It's now a protocol in `azure-core`, [SupportsGetToken](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/credentials.py), rather than an ABC.